### PR TITLE
[JsonPatch]: Added NotNull attribute to parameters

### PR DESCRIPTION
**Pull request by kirthik**
cc @rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs [hunk 1]
```diff
@@ -542,9 +551,9 @@ public void Copy(Operation<TModel> operation, TModel objectToApplyTo)
 
         private bool CheckIfPropertyExists(
             JsonPatchProperty patchProperty,
-            TModel objectToApplyTo,
-            Operation<TModel> operation,
-            string propertyPath)
+            [NotNull] TModel objectToApplyTo,
+            [NotNull] Operation<TModel> operation,
+            [NotNull] string propertyPath)
```
Comment (rynowak) [inline src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs]: We don't generally do `[NotNull]` on private methods. We want all of these things to be caught as close to the calling code as possible.

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs [hunk 2]
```diff
@@ -238,7 +244,7 @@ private void Add(string path, object value, TModel objectToApplyTo, Operation<TM
         /// </summary>
         /// <param name="operation">The move operation.</param>
         /// <param name="objectApplyTo">Object to apply the operation to.</param>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs]: `objectApplyTo` => `objectToApplyTo`

File: src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs [hunk 3]
```diff
@@ -323,7 +329,7 @@ public void Move(Operation<TModel> operation, TModel objectToApplyTo)
         /// </summary>
         /// <param name="operation">The remove operation.</param>
         /// <param name="objectApplyTo">Object to apply the operation to.</param>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs]: `objectApplyTo` => `objectToApplyTo`

File: src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs [hunk 4]
```diff
@@ -313,8 +321,8 @@ public JsonPatchDocument<TModel> Replace<TProp>(Expression<Func<TModel, IList<TP
         /// <param name="path"></param>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs]: Remove the line above this line.

File: src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs [hunk 5]
```diff
@@ -438,8 +445,8 @@ public JsonPatchDocument<TModel> Replace<TProp>(Expression<Func<TModel, IList<TP
...
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs]: Remove above line.

File: src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs [hunk 6]
```diff
@@ -46,7 +49,7 @@ public JsonPatchDocument(List<Operation<TModel>> operations, IContractResolver c
         /// <param name="path">path</param>
         /// <param name="value">value</param>
         /// <returns></returns>
```
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs]: Are all these `<returns>` intentionally left blank?

**Comment by ajaybhargavb** [comment]
:shipit: after minor fixes

File: src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs (hunk 6 above)
Comment (kirthik) [inline src/Microsoft.AspNet.JsonPatch/JsonPatchDocumentOfT.cs]: No. Added return type now.

**Comment by kirthik** [comment]
merged 38748b54eb7ef8e7cba2365d0621fcf4826dda4b

**Pull request closed without merging**